### PR TITLE
Feature/7260 add catalogue price tables

### PR DESCRIPTION
**Pull request by sam-roberts-bjss**
Story: [AB#7260](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7260)
Task: [AB#7518](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7518)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertTimeUnits.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿
+IF NOT EXISTS (SELECT * FROM dbo.TimeUnit)
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/PostDeployment/InsertTimeUnits.sql:2]: Starts on line 2, not line 1

File: src/NHSD.BuyingCatalogue.Database/Tables/CataloguePrice.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿CREATE TABLE dbo.CataloguePrice
+(
+    CataloguePriceId INT IDENTITY(1,1) NOT NULL,
+    CatalogueItemId NVARCHAR(14) NOT NULL,
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/CataloguePrice.sql:4]: Worth noting all of the other tables in BC have the types lower case e.g. varchar. Also nvarchar isnt widley used in BC (instead varchar is used). Do we want to keep it the same for consistency?

File: src/NHSD.BuyingCatalogue.Database/Tables/CataloguePrice.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    ProvisioningTypeId INT NOT NULL,
+    CataloguePriceTypeId INT NOT NULL,
+    PricingUnitId UNIQUEIDENTIFIER NOT NULL,
+    TimeUnitId INT NULL,
+    CurrencyCode NVARCHAR(3) NOT NULL,
+    LastUpdated DATETIME2(7) NOT NULL
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/CataloguePrice.sql:10]: There should be a comma at the end of here

File: src/NHSD.BuyingCatalogue.Database/Tables/CataloguePrice.sql [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+    CONSTRAINT PK_CataloguePrice PRIMARY KEY CLUSTERED (CataloguePriceId),
+    CONSTRAINT FK_CataloguePrice_ProvisioningType_ProvisioningTypeId FOREIGN KEY (ProvisioningTypeId) REFERENCES dbo.ProvisioningType(ProvisioningTypeId),
+    CONSTRAINT FK_CataloguePrice_CataloguePriceType_CataloguePriceTypeId FOREIGN KEY (CataloguePriceTypeId) REFERENCES dbo.CataloguePriceType(CataloguePriceTypeId),
+    CONSTRAINT FK_CataloguePrice_PricingUnit_PricingUnitId FOREIGN KEY (PricingUnitId) REFERENCES dbo.PricingUnit(PricingUnitId),
+    CONSTRAINT FK_CataloguePrice_TimeUnit_TimeUnitId FOREIGN KEY (TimeUnitId) REFERENCES dbo.TimeUnit(TimeUnitId)
+)
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/CataloguePrice.sql:17]: Need to add a semicolon

File: src/NHSD.BuyingCatalogue.Database/Tables/CataloguePriceFlat.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+﻿CREATE TABLE dbo.CataloguePriceFlat
+(
+    CataloguePriceId INT NOT NULL,
+    Price DECIMAL(18,3) NOT NULL
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/CataloguePriceFlat.sql:4]: comma is needed at the end of here

File: src/NHSD.BuyingCatalogue.Database/Tables/CataloguePriceFlat.sql [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+    CONSTRAINT PK_CataloguePriceFlat PRIMARY KEY NONCLUSTERED (CataloguePriceId),
+    CONSTRAINT FK_CataloguePriceFlat_CataloguePrice_CataloguePriceId FOREIGN KEY (CataloguePriceId) REFERENCES dbo.CataloguePrice(CataloguePriceId)
+)
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/CataloguePriceFlat.sql:7]: semicolon

File: src/NHSD.BuyingCatalogue.Database/Tables/CataloguePriceTier.sql [hunk 7]
```diff
@@ -0,0 +1,10 @@
+﻿CREATE TABLE dbo.CataloguePriceTier
+(
+    CataloguePriceTierId INT IDENTITY(1,1) NOT NULL,
+    CataloguePriceId INT NOT NULL,
+    BandStart INT NOT NULL,
+    BandEnd INT NULL,
+    Price DECIMAL(18,3) NOT NULL,
+    CONSTRAINT PK_CataloguePriceTier PRIMARY KEY NONCLUSTERED (CataloguePriceTierId),
+    CONSTRAINT FK_CataloguePriceTier_CataloguePrice_CataloguePriceId FOREIGN KEY (CataloguePriceId) REFERENCES dbo.CataloguePrice(CataloguePriceId)
+)
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/CataloguePriceTier.sql:10]: ;

File: src/NHSD.BuyingCatalogue.Database/Tables/CataloguePriceType.sql [hunk 8]
```diff
@@ -0,0 +1,6 @@
+﻿CREATE TABLE dbo.CataloguePriceType
+(
+    CataloguePriceTypeId INT NOT NULL,
+    [Name] NVARCHAR(10) NOT NULL
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/CataloguePriceType.sql:4]: ,

File: src/NHSD.BuyingCatalogue.Database/Tables/CataloguePriceType.sql [hunk 9]
```diff
@@ -0,0 +1,6 @@
...
+    CONSTRAINT PK_CataloguePriceType PRIMARY KEY CLUSTERED (CataloguePriceTypeId)
+)
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/CataloguePriceType.sql:6]: ;

File: src/NHSD.BuyingCatalogue.Database/Tables/PricingUnit.sql [hunk 10]
```diff
@@ -0,0 +1,8 @@
+﻿CREATE TABLE dbo.PricingUnit
+(
+    PricingUnitId UNIQUEIDENTIFIER NOT NULL,
+    [Name] NVARCHAR(20) NOT NULL,
+    TierName NVARCHAR(20) NOT NULL,
+    [Description] NVARCHAR(35) NOT NULL
+    CONSTRAINT PK_PricingUnit PRIMARY KEY NONClUSTERED (PricingUnitId)
+)
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/PricingUnit.sql:8]: ;

File: src/NHSD.BuyingCatalogue.Database/Tables/PricingUnit.sql [hunk 11]
```diff
@@ -0,0 +1,8 @@
+﻿CREATE TABLE dbo.PricingUnit
+(
+    PricingUnitId UNIQUEIDENTIFIER NOT NULL,
+    [Name] NVARCHAR(20) NOT NULL,
+    TierName NVARCHAR(20) NOT NULL,
+    [Description] NVARCHAR(35) NOT NULL
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/PricingUnit.sql:6]: ,

File: src/NHSD.BuyingCatalogue.Database/Tables/ProvisioningType.sql [hunk 12]
```diff
@@ -0,0 +1,6 @@
+﻿CREATE TABLE dbo.ProvisioningType
+(
+    ProvisioningTypeId INT NOT NULL,
+    [Name] NVARCHAR(35) NOT NULL
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/ProvisioningType.sql:4]: ,

File: src/NHSD.BuyingCatalogue.Database/Tables/ProvisioningType.sql [hunk 13]
```diff
@@ -0,0 +1,6 @@
...
+    CONSTRAINT PK_ProvisioningType PRIMARY KEY CLUSTERED (ProvisioningTypeId)
+)
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/ProvisioningType.sql:6]: ;

File: src/NHSD.BuyingCatalogue.Database/Tables/TimeUnit.sql [hunk 14]
```diff
@@ -0,0 +1,7 @@
+﻿CREATE TABLE dbo.TimeUnit
+(
+    TimeUnitId INT NOT NULL,
+    [Name] NVARCHAR(20) NOT NULL,
+    [Description] NVARCHAR(32) NOT NULL
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/TimeUnit.sql:5]: ,

File: src/NHSD.BuyingCatalogue.Database/Tables/TimeUnit.sql [hunk 15]
```diff
@@ -0,0 +1,7 @@
...
+    CONSTRAINT PK_TimeUnit PRIMARY KEY CLUSTERED (TimeUnitId)
+)
```
Comment (RuthBerry96) [inline src/NHSD.BuyingCatalogue.Database/Tables/TimeUnit.sql:7]: ;

File: src/NHSD.BuyingCatalogue.Database/Tables/CataloguePrice.sql (hunk 2 above)
Comment (sam-roberts-bjss) [reply src/NHSD.BuyingCatalogue.Database/Tables/CataloguePrice.sql]: OK will change them.

File: src/NHSD.BuyingCatalogue.Database/Tables/CataloguePrice.sql (hunk 3 above)
Comment (sam-roberts-bjss) [reply src/NHSD.BuyingCatalogue.Database/Tables/CataloguePrice.sql]: The project does compile without the comma but added them.

**Pull request merged**